src/services/completions.ts

```diff
@@ -1062,6 +1055,135 @@ namespace ts.Completions {
         return undefined;
     }
 
+    function getEntryForObjectLiteralMethodCompletion(

```

Comment:
Ok, so the main thing I’m wondering is if we can consolidate a few of the separate steps here. My understanding of the process is something like this:

1. When processing symbols in `tryGetObjectLikeCompletionSymbols`, you do a fast check via `isObjectLiteralMethodSymbol` and as long as that passes, you push the symbol and an origin (with no info other than `kind`) onto the `symbols` array.

2. After pushing all the symbols on, you update their SortText—I can’t quite tell why this can’t be done in (1).

3. At the very end of the completions process, in `createCompletionEntry`, you call into this function here for any symbols that got their origin set accordingly in (1). This function does all the hard work and introduces some extra conditions which might result in us dropping the entry altogether.

I could definitely be missing something, but I don’t see much of a reason for these three steps to be separate. It seems like you could go as far as creating the `MethodDeclaration` where you currently are only doing step (1), and then store that on the origin. That way, if you can’t create the MethodDeclaration, you don’t need to bother with setting SortText. In other words, by the time you push a symbol and an origin onto those arrays, you know for sure it will be able to turn into a CompletionEntry because most of the work to do it is already done. It seems like this would consolidate the implementation a little. Does that make sense, or what did I miss?

Reply:
That made sense. I updated the code so that (3) is now done in (1).

About (2), we can't do that when pushing the method snippet symbols, because we need all of the object completions to have the transformed format. Still, I moved the functions that set/transform sort texts to avoid doing some checks.

Reply:
Makes sense about (2). Looks much better now without having to create the Set of seen symbols 👍